src/app/Marine2/components/ui/InputLimitSelector/InputLimitSelector.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from "react"
+import { useInputLimit, useInputLimitSelector } from "@elninotech/mfd-modules"
+import Button from "../Button"
+import DeviceSettingModal from "../DeviceSettingModal"
+import { translate } from "react-i18nify"
+import classnames from "classnames"
+import { observer } from "mobx-react"
+
+const InputLimitSelector = ({ inputId }: Props) => {
+  const USAmperage = [10, 15, 20, 30, 50, 100]
+  const EUAmperage = [3, 6, 10, 13, 16, 25, 32, 63]
+
+  /**
+   * - Mask the Product id with `0xFF00`
+   * - If the result is `0x1900` or `0x2600` it is an EU model (230VAC)
+   * - If the result is `0x2000` or `0x2700` it is a US model (120VAC)
+   */
+
+  const getSuggestedAmperageValuesList = (productId: number) => {
+    const result = productId & 0xff00
+    if (result === 0x1900 || result === 0x2600) {
+      return EUAmperage
+    } else if (result === 0x2000 || result === 0x2700) {
+      return USAmperage
+    } else {
+      console.log(`Could not determine amperage US/EU for product id ${productId}`)
+      return USAmperage
+    }
+  }
+
+  const { currentLimitIsAdjustable } = useInputLimit(inputId)
+  const { currentLimit, currentLimitMax, productId, updateLimit } = useInputLimitSelector(inputId)
+  const amperageList = getSuggestedAmperageValuesList(productId).filter((value) => {
+    return value <= (currentLimitMax ?? 100)
+  })
+
+  const [limitForSubmission, setLimitForSubmission] = useState(Number(currentLimit))
+
+  const [isLimitModalOpen, setIsLimitModalOpen] = useState(false)
+
+  useEffect(() => {
+    setLimitForSubmission(Number(currentLimit))
+  }, [currentLimit])
+
+  const increaseLimit = () => {
+    if (limitForSubmission < (currentLimitMax ?? 100)) setLimitForSubmission(limitForSubmission + 1)
+  }
+
+  const decreaseLimit = () => {
+    if (limitForSubmission > 0) setLimitForSubmission(limitForSubmission - 1)
+  }
+
+  const closeLimitModal = () => {
+    setIsLimitModalOpen(false)
+    setLimitForSubmission(Number(currentLimit))
+  }
+
+  const submitLimit = () => {
+    updateLimit(limitForSubmission)
+    closeLimitModal()
+  }
+
+  if (!currentLimitIsAdjustable) return null
+  return (
+    <>
+      <Button className="w-full mr-4" size="md" onClick={() => setIsLimitModalOpen(!isLimitModalOpen)}>
+        {!!currentLimit ? Number(currentLimit) + "A" : 0 + "A"}
+      </Button>
+      <DeviceSettingModal open={isLimitModalOpen} onClose={closeLimitModal} onSet={submitLimit} width={"lg"}>
+        <label className="flex w-full justify-center text-lg mb-3">{translate("devices.Input current limit")}</label>
+        <div className="flex flex-row justify-center mt-10">
```

Comment:
flex direction is row by default

Reply:
That's true, but I also prefer to explicitly set `flex-row` to prevent possible defaults changes and also for better consistency while reading the code